CMakeLists.txt

```diff
@@ -0,0 +1,691 @@
+    message(STATUS "CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
+endif()
+
+# Set C++ standard
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Find Python
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
+
+# Include scikit-build-core helpers
+if(SKBUILD)
+    find_package(pybind11 CONFIG REQUIRED)
+endif()
+
+# Platform-specific settings
+if(WIN32)
+    set(LIB_EXT "dll")
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+else()
+    set(LIB_EXT "so")
+endif()
+
+# Compiler-specific settings
+if(MSVC)
+    set(EXTRA_CXX_FLAGS "/permissive-" "/W4")
+    # Disable automatic manifest generation
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /MANIFEST:NO")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /MANIFEST:NO")
+else()
+    set(BASE_WARNS
+        "-Wall" "-Wextra" "-Wcast-qual" "-Wwrite-strings" "-Wconversion"
+        "-Wno-missing-field-initializers" "-Werror=shadow"
+    )
+    set(CXX_WARNS ${BASE_WARNS} "-Wnon-virtual-dtor" "-Woverloaded-virtual")
+    set(EXTRA_CXX_FLAGS
+        "-std=c++11" "-fvisibility=hidden" "-fvisibility-inlines-hidden"
+        ${CXX_WARNS}
+    )
+    if(NOT WIN32)
+        list(APPEND EXTRA_CXX_FLAGS "-flto")
+    endif()
+    
+    # Windows-specific GCC/MinGW settings
+    if(WIN32)
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--exclude-all-symbols")
+    else()
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -flto")
+        
+        # Linux-specific: avoid libstdc++ runtime dependency
+        if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+            set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -static-libstdc++")
+        endif()
+    endif()
+endif()
+
+# Common definitions
+set(EXTRA_DEFINES "__STDC_FORMAT_MACROS")
+
+# Get Python library for linking
+if(APPLE)
+    set(PYTHON_LIB_VAR "LIBRARY")
+else()
+    set(PYTHON_LIB_VAR "LDLIBRARY")
+endif()
+
+execute_process(
+    COMMAND ${Python_EXECUTABLE} -c "import sysconfig; print(sysconfig.get_config_var('${PYTHON_LIB_VAR}') or '')"
+    OUTPUT_VARIABLE PYTHON_LDLIBRARY
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+if(PYTHON_LDLIBRARY)
+    get_filename_component(PYTHON_LIB_LINK ${PYTHON_LDLIBRARY} NAME_WLE)
+    string(REGEX REPLACE "^lib" "" PYTHON_LIB_LINK ${PYTHON_LIB_LINK})
+else()
+    string(REPLACE "." "" PYTHON_VERSION_NODOT ${Python_VERSION_MAJOR}${Python_VERSION_MINOR})
+    set(PYTHON_LIB_LINK "python${PYTHON_VERSION_NODOT}")
+endif()
+
+if(WIN32)
+    set(PYTHON_LIB "${PYTHON_LIB_LINK}.${LIB_EXT}")
+elseif(APPLE)
+    set(PYTHON_LIB_DIR ${Python_STDLIB}/../..)
+    if(EXISTS "${PYTHON_LIB_DIR}/lib${PYTHON_LIB_LINK}.dylib")
+        set(PYTHON_LIB "${PYTHON_LIB_DIR}/lib${PYTHON_LIB_LINK}.dylib")
+    else()
+        set(PYTHON_LIB "${PYTHON_LIB_DIR}/lib${PYTHON_LIB_LINK}.so")
+    endif()
+else()
+    set(PYTHON_LIB "lib${PYTHON_LIB_LINK}.${LIB_EXT}")
+endif()
```

Comment:
I'm fine with deleting all this and the bits from embed.cpp. I was planning on doing that in the library refactor.